plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/AlteredFiles.java

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.git.client.compare;
+
+import org.eclipse.che.ide.api.resources.Project;
+import org.eclipse.che.ide.ext.git.client.compare.FileStatus.Status;
+import org.eclipse.che.ide.util.loging.Log;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.eclipse.che.ide.ext.git.client.compare.FileStatus.defineStatus;
+
+/**
+ * Describes changed in any way files in git comparison process.
+ *
+ * @author Mykola Morhun
+ */
+public class AlteredFiles {
+
+    private final Project                       project;
+    private final LinkedHashMap<String, Status> alteredFilesStatuses;
+    private final List<String>                  alteredFilesList;
+
+    /**
+     * Parses raw git diff string and creates advanced representation.
+     *
+     * @param project
+     *         the project under diff operation
+     * @param diff
+     *         plain result of git diff operation
+     */
+    public AlteredFiles(Project project, String diff) {
+        this.project = project;
+
+        alteredFilesStatuses = new LinkedHashMap<>();
+        if (!isNullOrEmpty(diff)) {
+            for (String item : diff.split("\n")) {
+                if (item.length() < 3 || item.charAt(1) != '\t') {
+                    throw new IllegalArgumentException("Invalid git diff format.");
```

Comment:
FYI: `git apply` command fails if file not found and applies patch if not a critical error found.